src/hotspot/cpu/ppc/c1_LIRAssembler_ppc.cpp

```diff
@@ -157,9 +157,9 @@ void LIR_Assembler::osr_entry() {
               mo = frame_map()->address_for_monitor_object(i);
       assert(ml.index() == noreg && mo.index() == noreg, "sanity");
       __ ld(R0, slot_offset + 0, OSR_buf);
-      __ std(R0, ml.disp(), ml.base());
+      __ std(R0, ml, noreg);
```

Comment:
No, `tmp=noreg` is default case.

So 
```suggestion
      __ std(R0, ml);
```
should work.